### ISSUE TITLE
The last build to work on Forge 1890 and Sponge 1509 is build 4.

### ISSUE THREAD

**Issue by inklit** [MinecraftPortCentral/GriefPrevention#78]
The last build to work on Forge 1890 and Sponge 1509-1890 is build 4.

Any newer builds and GriefPrevention either fails to load completely, or commands work but there is no claims and the shovel does not work at all so new claims are unable to be created.

I will setup a testing environment to do more experiments to see if I can figure out why this happens, or at least find some log record that sheds some light.

In the mean time, please look into this.

**Comment by bloodmc** [comment]
@inklit You need to paste the full server log. I'm running latest builds perfectly fine. Make sure you have PermissionsEx installed.

**Comment by inklit** [comment]
That is probably the issue, we are using PermissionManager instead.

Is there any chance you could make it compatible with PermissionManager as well?

**Comment by bloodmc** [comment]
@inklit paste the full server log.

**Comment by inklit** [comment]
Okay, let me mirror my server real quick so I don't have to do this live.

**Comment by inklit** [comment]
http://javaprophet.com/fml-server-latest1.log

**Comment by bloodmc** [comment]
OK so this is the reason why I require PEX since it has a default user called "default" which essentially allows me to set permissions for ALL users.

https://github.com/MinecraftPortCentral/GriefPrevention/blob/6b8c08c19f5806299fc70105db3d15af7821ac9d/src/me/ryanhamshire/griefprevention/GriefPrevention.java#L289

PermissionsManager would have to support this as well for GP to work.

@zml2008 Can you chime in and let us know how PermissionManager would add support for the line above.

**Comment by inklit** [comment]
I've tested with PermissionsEx and it does work. This issue can be closed unless you wish to fix keep it open until the permission system issue is resolved.

**Comment by zml2008** [comment]
Well in API 5 there are new methods added to PermissionService and SubjectCollection that return the appropriate defaults Subject, which would replace that line. PermissionManager would then implement those subjects. However, PM also fails to fully implement contexts and custom subject types, which have been a core part of the permissions API since the beginning, so GP would still not work.

tl;dr PermissionsManager doesn't implement the permissions api beyond the most basic features so GP can't work with it.

**Comment by bloodmc** [comment]
@zml2008 Thanks for the information!

@inklit As you can see from zml's response, PermissionManager does not implement contexts which GriefPrevention makes use of fully. You will have to switch to PEX if you want to use GriefPrevention. If PermissionManager decides to implement contexts in the future and GP has issues, I'll look at it then but as of now it doesn't so closing ticket.
